cmd/kubeadm/app/cmd/reset.go

```diff
@@ -262,3 +276,32 @@ func resetConfigDir(configPathDir, pkiPathDir string) {
 		}
 	}
 }
+
+// promptIfMasterNode will prompt the user for confirmation when running reset on a master node
+func promptIfMasterNode(in io.Reader, resetAlreadyConfirmed bool, staticPodDir string) error {
+	if resetAlreadyConfirmed {
+		return nil
+	}
+	// determine if this is a master node by checking for the existence of the api server static pod manifest
+	apiServerManifest := kubeadmconstants.GetStaticPodFilepath(kubeadmconstants.KubeAPIServer, staticPodDir)
+	_, err := os.Stat(apiServerManifest)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("could not determine if this is a master node: %v", err)
+	}
+
+	// This is a master node - prompt the user for confirmation
+	fmt.Printf("[Warning] Are you sure you want to reset this master node? Type the word %q to continue: ", confirmationWord)
```

Comment:
I'd prefer just a [y/n] response here.  Also we might want to check is kubectl has any blocking interactions and copy the pattern.  

If they do I would be surprised if there wasn't already a utility for it. 

/cc @pwittrock

Reply:
Sounds good. I'll look to see if there are blocking interactions in kubectl that we can be consistent with.